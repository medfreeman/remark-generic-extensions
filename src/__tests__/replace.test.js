import { transformToHtml } from "./helpers/transform";

describe("replace inline extension", () => {
  test("should add an extension", () =>
    expect(
      transformToHtml("!icon", {
        elements: {
          icon: {
            replace: function(type) {
              return {
                type,
                data: {
                  hName: "icon",
                  hChildren: [
                    {
                      type: "text",
                      value: "¬|¬"
                    }
                  ],
                  hProperties: {}
                }
              };
            }
          }
        }
      })
    ).toMatchSnapshot());
});

describe("replace block extension", () => {
  test("should add an extension", () =>
    expect(
      transformToHtml(
        `icon:
:::
:::`,
        {
          elements: {
            icon: {
              replace: function(type) {
                return {
                  type,
                  data: {
                    hName: "icon",
                    hChildren: [
                      {
                        type: "text",
                        value: "¬|¬"
                      }
                    ],
                    hProperties: {}
                  }
                };
              }
            }
          }
        }
      )
    ).toMatchSnapshot());
});
