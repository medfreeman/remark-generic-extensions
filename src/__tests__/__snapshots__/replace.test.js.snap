// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace block extension should add an extension 1`] = `<icon>¬|¬</icon>`;

exports[`replace inline extension should add an extension 1`] = `
<p>
  <icon>¬|¬</icon>
</p>
`;
